### PR TITLE
[ossia-score] Update to 3.2.0, reinstate clang arm64

### PR DESCRIPTION
**Pull request by jcelerier**
Also this uses more dependencies from the host environment as requested.

### CODE REVIEW

**Comment by jcelerier** [comment]
@Biswa96 the MINGW64 build fails because of connection failure during dependencies download but I do not have the rights to restart it:

```

  :: Retrieving packages...
  error: failed retrieving file 'mingw-w64-x86_64-icu-74.2-1-any.pkg.tar.zst.sig' from mirror.umd.edu : Operation too slow. Less than 1 bytes/sec transferred the last 10 seconds
  error: failed retrieving file 'mingw-w64-x86_64-qt6-base-6.7.0-3-any.pkg.tar.zst.sig' from mirror.umd.edu : Operation too slow. Less than 1 bytes/sec transferred the last 10 seconds
   mingw-w64-x86_64-python-3.11.9-1-any downloading...
   mingw-w64-x86_64-icu-74.2-1-any downloading...
   mingw-w64-x86_64-ffmpeg-6.1.1-10-any downloading...
   mingw-w64-x86_64-qt6-base-6.7.0-3-any downloading...
   mingw-w64-x86_64-qt6-declarative-6.7.0-1-any downloading...
  error: failed retrieving file 'mingw-w64-x86_64-python-3.11.9-1-any.pkg.tar.zst.sig' from mirror.umd.edu : Operation too slow. Less than 1 bytes/sec transferred the last 10 seconds
  warning: too many errors from mirror.umd.edu, skipping for the remainder of this transaction
  error: failed retrieving file 'mingw-w64-x86_64-ffmpeg-6.1.1-10-any.pkg.tar.zst' from mirror.msys2.org : Operation too slow. Less than 1 bytes/sec transferred the last 10 seconds
  error: failed retrieving file 'mingw-w64-x86_64-qt6-declarative-6.7.0-1-any.pkg.tar.zst' from mirror.msys2.org : Operation too slow. Less than 1 bytes/sec transferred the last 10 seconds
```

File: mingw-w64-ossia-score/PKGBUILD [hunk 1]
```diff
@@ -49,6 +56,11 @@ build() {
     extra_config+=("-DCMAKE_BUILD_TYPE=Debug")
   fi
 
+  CXX_ADDITIONAL_FLAGS=
+  if c++ --version | grep clang; then
```
[the remark this answers is not in the document]
Comment (jcelerier) [reply mingw-w64-ossia-score/PKGBUILD]: ok then, i'll follow established practice :)

**Comment by ognevny** [comment]
> ok, should be good to merge :)

I'll wait until CI is green

**Comment by jcelerier** [comment]
should be good !

**Comment by jcelerier** [comment]
is it normal that there's no clang arm64 CI though ? I explicitly reenabled it in the PKGBUILD

**Comment by ognevny** [comment]
oh no! I should have chose squash option

**Comment by jcelerier** [comment]
thx !

**Comment by jcelerier** [comment]
I can push a revert but it'll make the history even messier

**Comment by ognevny** [comment]
> I can push a revert but it'll make the history even messier

that's not so bad though. that's my bad

**Comment by jcelerier** [comment]
>      yes, for now ARM runners can be self-hosted only

hmm.. for this project I already use github-hosted aarch64 macOS runners (on which you can easily spin a Linux aarch64 docker that will build mingw stuff I guess?)

**Comment by ognevny** [comment]
> hmm.. for this project I already use github-hosted aarch64 macOS runners (on which you can easily spin a Linux aarch64 docker that will build mingw stuff I guess?). They have been available for a couple months.

I'm not a maintainer of arm machine that is used there. I can't tell you anything

**Comment by MehdiChinoune** [comment]
@ognevny Could you please stash commits instead of merging them like that. At least ask them to write better commit messages.

**Comment by jcelerier** [comment]
sorry, I committed assuming it would be stashed at the end. Next time i'll push --force my own branch instead

**Comment by ognevny** [comment]
failed for clangarm64 with

<details>

```
[112/338] Building CXX object src/plugins/score-plugin-gfx/Spout/CMakeFiles/Spout.dir/Unity/unity_0_cxx.cxx.obj
  FAILED: src/plugins/score-plugin-gfx/Spout/CMakeFiles/Spout.dir/Unity/unity_0_cxx.cxx.obj 
  C:\msys64\clangarm64\bin\clang++.exe @src\plugins\score-plugin-gfx\Spout\CMakeFiles\Spout.dir\Unity\unity_0_cxx.cxx.obj.rsp -MD -MT src/plugins/score-plugin-gfx/Spout/CMakeFiles/Spout.dir/Unity/unity_0_cxx.cxx.obj -MF src\plugins\score-plugin-gfx\Spout\CMakeFiles\Spout.dir\Unity\unity_0_cxx.cxx.obj.d -o src/plugins/score-plugin-gfx/Spout/CMakeFiles/Spout.dir/Unity/unity_0_cxx.cxx.obj -c C:/M/B/src/build-CLANGARM64/src/plugins/score-plugin-gfx/Spout/CMakeFiles/Spout.dir/Unity/unity_0_cxx.cxx
  clang++: warning: argument unused during compilation: '-msse4' [-Wunused-command-line-argument]
  In file included from C:/M/B/src/build-CLANGARM64/src/plugins/score-plugin-gfx/Spout/CMakeFiles/Spout.dir/Unity/unity_0_cxx.cxx:4:
  In file included from C:/M/B/src/ossia-score-3.2.0/3rdparty/Spout/Spout.cpp:247:
  In file included from C:/M/B/src/ossia-score-3.2.0/3rdparty/Spout/Spout.h:35:
  In file included from C:/M/B/src/ossia-score-3.2.0/3rdparty/Spout/SpoutGL.h:41:
  In file included from C:/M/B/src/ossia-score-3.2.0/3rdparty/Spout/SpoutCopy.h:41:
  C:/msys64/clangarm64/lib/clang/18/include/emmintrin.h:14:2: error: "This header is only meant to be used on x86 and x64 architecture"
     14 | #error "This header is only meant to be used on x86 and x64 architecture"
        |  ^
  In file included from C:/M/B/src/build-CLANGARM64/src/plugins/score-plugin-gfx/Spout/CMakeFiles/Spout.dir/Unity/unity_0_cxx.cxx:4:
  In file included from C:/M/B/src/ossia-score-3.2.0/3rdparty/Spout/Spout.cpp:247:
  In file included from C:/M/B/src/ossia-score-3.2.0/3rdparty/Spout/Spout.h:35:
  In file included from C:/M/B/src/ossia-score-3.2.0/3rdparty/Spout/SpoutGL.h:41:
  In file included from C:/M/B/src/ossia-score-3.2.0/3rdparty/Spout/SpoutCopy.h:41:
  In file included from C:/msys64/clangarm64/lib/clang/18/include/emmintrin.h:17:
  C:/msys64/clangarm64/lib/clang/18/include/xmmintrin.h:14:2: error: "This header is only meant to be used on x86 and x64 architecture"
     14 | #error "This header is only meant to be used on x86 and x64 architecture"
        |  ^
  In file included from C:/M/B/src/build-CLANGARM64/src/plugins/score-plugin-gfx/Spout/CMakeFiles/Spout.dir/Unity/unity_0_cxx.cxx:4:
  In file included from C:/M/B/src/ossia-score-3.2.0/3rdparty/Spout/Spout.cpp:247:
  In file included from C:/M/B/src/ossia-score-3.2.0/3rdparty/Spout/Spout.h:35:
  In file included from C:/M/B/src/ossia-score-3.2.0/3rdparty/Spout/SpoutGL.h:41:
  In file included from C:/M/B/src/ossia-score-3.2.0/3rdparty/Spout/SpoutCopy.h:41:
  In file included from C:/msys64/clangarm64/lib/clang/18/include/emmintrin.h:17:
  In file included from C:/msys64/clangarm64/lib/clang/18/include/xmmintrin.h:17:
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:14:2: error: "This header is only meant to be used on x86 and x64 architecture"
     14 | #error "This header is only meant to be used on x86 and x64 architecture"
        |  ^
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:39:3: error: use of undeclared identifier '__builtin_ia32_emms'; did you mean '__builtin_isless'?
     39 |   __builtin_ia32_emms();
        |   ^
  C:/msys64/clangarm64/include/c++/v1/__math/traits.h:145:10: note: '__builtin_isless' declared here
    145 |   return __builtin_isless((type)__x, (type)__y);
        |          ^
  In file included from C:/M/B/src/build-CLANGARM64/src/plugins/score-plugin-gfx/Spout/CMakeFiles/Spout.dir/Unity/unity_0_cxx.cxx:4:
  In file included from C:/M/B/src/ossia-score-3.2.0/3rdparty/Spout/Spout.cpp:247:
  In file included from C:/M/B/src/ossia-score-3.2.0/3rdparty/Spout/Spout.h:35:
  In file included from C:/M/B/src/ossia-score-3.2.0/3rdparty/Spout/SpoutGL.h:41:
  In file included from C:/M/B/src/ossia-score-3.2.0/3rdparty/Spout/SpoutCopy.h:41:
  In file included from C:/msys64/clangarm64/lib/clang/18/include/emmintrin.h:17:
  In file included from C:/msys64/clangarm64/lib/clang/18/include/xmmintrin.h:17:
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:39:23: error: too few arguments to function call, expected 2, have 0
     39 |   __builtin_ia32_emms();
        |   ~~~~~~~~~~~~~~~~~~~~^
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:56:19: error: use of undeclared identifier '__builtin_ia32_vec_init_v2si'
     56 |     return (__m64)__builtin_ia32_vec_init_v2si(__i, 0);
        |                   ^
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:73:12: error: use of undeclared identifier '__builtin_ia32_vec_ext_v2si'
     73 |     return __builtin_ia32_vec_ext_v2si((__v2si)__m, 0);
        |            ^
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:135:19: error: use of undeclared identifier '__builtin_ia32_packsswb'
    135 |     return (__m64)__builtin_ia32_packsswb((__v4hi)__m1, (__v4hi)__m2);
        |                   ^
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:165:19: error: use of undeclared identifier '__builtin_ia32_packssdw'
    165 |     return (__m64)__builtin_ia32_packssdw((__v2si)__m1, (__v2si)__m2);
        |                   ^
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:195:19: error: use of undeclared identifier '__builtin_ia32_packuswb'
    195 |     return (__m64)__builtin_ia32_packuswb((__v4hi)__m1, (__v4hi)__m2);
        |                   ^
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:222:19: error: use of undeclared identifier '__builtin_ia32_punpckhbw'
    222 |     return (__m64)__builtin_ia32_punpckhbw((__v8qi)__m1, (__v8qi)__m2);
        |                   ^
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:245:19: error: use of undeclared identifier '__builtin_ia32_punpckhwd'
    245 |     return (__m64)__builtin_ia32_punpckhwd((__v4hi)__m1, (__v4hi)__m2);
        |                   ^
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:266:19: error: use of undeclared identifier '__builtin_ia32_punpckhdq'
    266 |     return (__m64)__builtin_ia32_punpckhdq((__v2si)__m1, (__v2si)__m2);
        |                   ^
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:293:19: error: use of undeclared identifier '__builtin_ia32_punpcklbw'
    293 |     return (__m64)__builtin_ia32_punpcklbw((__v8qi)__m1, (__v8qi)__m2);
        |                   ^
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:316:19: error: use of undeclared identifier '__builtin_ia32_punpcklwd'
    316 |     return (__m64)__builtin_ia32_punpcklwd((__v4hi)__m1, (__v4hi)__m2);
        |                   ^
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:337:19: error: use of undeclared identifier '__builtin_ia32_punpckldq'
    337 |     return (__m64)__builtin_ia32_punpckldq((__v2si)__m1, (__v2si)__m2);
        |                   ^
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:358:19: error: use of undeclared identifier '__builtin_ia32_paddb'; did you mean '__builtin_arm_addg'?
    358 |     return (__m64)__builtin_ia32_paddb((__v8qi)__m1, (__v8qi)__m2);
        |                   ^
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:358:19: note: '__builtin_arm_addg' declared here
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:358:19: error: first argument of MTE builtin function must be a pointer ('__v8qi' (vector of 8 'char' values) invalid)
    358 |     return (__m64)__builtin_ia32_paddb((__v8qi)__m1, (__v8qi)__m2);
        |                   ^                    ~~~~~~~~~~~~
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:379:19: error: use of undeclared identifier '__builtin_ia32_paddw'; did you mean '__builtin_arm_addg'?
    379 |     return (__m64)__builtin_ia32_paddw((__v4hi)__m1, (__v4hi)__m2);
        |                   ^
  C:/msys64/clangarm64/lib/clang/18/include/mmintrin.h:358:19: note: '__builtin_arm_addg' declared here
  
    358 |     return (__m64)__builtin_ia32_paddb((__v8qi)__m1, (__v8qi)__m2);
  
        |                   ^
```

</details>

**Comment by Biswa96** [comment]
> failed for clangarm64 with

It could be fixed by not including the headers for arm here https://github.com/ossia/score/blob/a96d6301131372e592b7b3ea295d1df675c2a49c/3rdparty/Spout/SpoutCopy.h#L41

**Comment by ognevny** [comment]
feel free to propose the change :)

**Comment by jcelerier** [comment]
hmm yeah I guess this library just does not work on ARM64. I'll add a flag to disable it in the next release.

**Comment by jcelerier** [comment]
okay, this feature will only be available on x86 now

**Pull request merged**